File: Inversion/serial/NA/src/hv_subs/PROGRAMS.330/CALPLOT/testf/tstcur.f
c---------------------------------------------------------------------c
c                                                                     c
c      COMPUTER PROGRAMS IN SEISMOLOGY                                c
c      VOLUME I                                                       c
c                                                                     c
c      PROGRAM: TSTCUR                                                c
c                                                                     c
c      COPYRIGHT (C)  1986 R. B. Herrmann                             c
c                                                                     c
c      Department of Earth and Atmospheric Sciences                   c
c      Saint Louis University                                         c
c      221 North Grand Boulevard                                      c
c      St. Louis, Missouri 63103                                      c
c      U. S. A.                                                       c
c                                                                     c
c---------------------------------------------------------------------c
	real*4 x(22),y(22)
	character*2 ic
	character*10 ttlx,ttly
	character*40 outstr
	call pinitf('INTER')
	call gunit('in')
	n = 20
	do 100 i=1,n
		x(i) = i
		y(i) = i*i*i
  100	continue
	do 1000 i=1,4
		call plot(2.0,1.0,-3)
		if(i.eq.1)then
			nocx=2
			nocy=3
			fct = 1.0
			inteq = 0
			lintyp = 0
		elseif(i.eq.2)then
			nocx=0
			nocy=2
			fct = 0.8
			inteq = i
			lintyp = -1
		elseif(i.eq.3)then
			nocy=0
			nocx=2
			fct = 1.0
			inteq = 1
			lintyp = +1
		elseif(i.eq.4)then
			nocx=0
			nocy=0
			fct = 0.8
			inteq = i
			lintyp = 0
		endif
		call factor(fct)
		yaxlen=6.0
		xaxlen=5.0
		call pltscl(x,xaxlen,n,x1,deltax,nocx)
		call pltscl(y,yaxlen,n,y1,deltay,nocy)
		ttlx = 'X-AXIS'
		mtx = 6
		ttly = 'Y-AXIS'
		mty = 6
		call algaxe(xaxlen,yaxlen,nocx,nocy,ttlx,ttly,mtx,mty,
     1		x1,y1,deltax,deltay)
		if(i.le.2)then
		call pltlog(x,y,n,x1,y1,deltax,deltay,lintyp,inteq,ht,
     1		nocx,nocy)
		else
		call pltlgd(x,y,n,x1,y1,deltax,deltay,lintyp,inteq,ht,
     1		nocx,nocy,8*i -1 , 0.10)
		endif
		call curuxy(xx,yy,x1,y1,deltax,deltay,nocx,nocy,ic)
		write(outstr,'(2f10.3,1x,a)')xx,yy,ic
		call gwrtxt(0.0,5.0,outstr,0)
		call factor(1.0)
		call plot(-2.0,-1.0,-3)
		call frame()
 1000	continue
	call pend()
	stop
	end

File: Inversion/parallel/NA/src/hv_subs/PROGRAMS.330/CALPLOT/testc/tstcur.c
/*
c---------------------------------------------------------------------c
c                                                                     c
c      COMPUTER PROGRAMS IN SEISMOLOGY                                c
c      VOLUME I                                                       c
c                                                                     c
c      PROGRAM: TSTCUR                                                c
c                                                                     c
c      COPYRIGHT (C)  1986 R. B. Herrmann                             c
c                                                                     c
c      Department of Earth and Atmospheric Sciences                   c
c      Saint Louis University                                         c
c      221 North Grand Boulevard                                      c
c      St. Louis, Missouri 63103                                      c
c      U. S. A.                                                       c
c                                                                     c
c---------------------------------------------------------------------c
*/

#include <math.h>
#include "calplot.h"


main()
{
	float x[22],y[22];
	char	ic[2], is[2];
	char	ttlx[10],ttly[10];
	char 	outstr[40];
	float fct, xaxlen, yaxlen, xx, yy, x1, y1, deltax, deltay, ht;
	int i, n;
	int nocx, nocy, inteq, lintyp, mtx, mty;

	ginitf("INTER","TSTCUR");
	gunit("in");
	n = 20;
	ht = 0.07;
	for(i=1;i<=n;i++){
		x[i-1] = i;
		y[i-1] = i*i*i;
	}
	for(i=1;i<=4;i++){
		plot(2.0,1.0,-3);
		if(i == 1){
			nocx=2;
			nocy=3;
			fct = 1.0;
			inteq = 0;
			lintyp = 0;
		} else if(i == 2){
			nocx=0;
			nocy=2;
			fct = 0.8;
			inteq = i;
			lintyp = -1;
		} else if(i == 3){
			nocy=0;
			nocx=2;
			fct = 1.0;
			inteq = 1;
			lintyp =  1;
		} else if(i == 4){
			nocx=0;
			nocy=0;
			fct = 0.8;
			inteq = i;
			lintyp = 0;
		}
		is[0] = (char)inteq;
		factor(fct);
		yaxlen=6.0;
		xaxlen=5.0;
		pltscl(x,xaxlen,n,&x1,&deltax,nocx);
		pltscl(y,yaxlen,n,&y1,&deltay,nocy);
		strcpy(ttlx , "X-AXIS");
		mtx = 6;
		strcpy(ttly , "Y-AXIS");
		mty = 6;
		algaxe(xaxlen,yaxlen,nocx,nocy,ttlx,ttly,mtx,mty,
			x1,y1,deltax,deltay);
		if(i <= 2){
			pltlog(x,y,n,x1,y1,deltax,deltay,lintyp,inteq,ht,
				nocx,nocy);
		} else {
			pltlgd(x,y,n,x1,y1,deltax,deltay,lintyp,inteq,ht,
				nocx,nocy,8*i -1 , 0.10);
		}
		curuxy(&xx,&yy,x1,y1,deltax,deltay,nocx,nocy,ic);
		sprintf(outstr,"%10.3f%10.3f %s",xx,yy,ic);
		gwrtxt(0.0,5.0,outstr,0);
		factor(1.0);
		plot(-2.0,-1.0,-3);
		frame();
	}
	pend();
}
